### ISSUE TITLE
prefer_self_in_static_references False positive in typealias 

### ISSUE THREAD

**Issue by la-pieuvre** [realm/SwiftLint#5009]
### New Issue Checklist

- [x ] Updated SwiftLint to the latest version
- [ x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

SwiftLint replace the type of the class by self in a typealias generating an error

```swift
class AClassWithAVeryLongName{
    typealias ACWAVLN =  AClassWithAVeryLongName
}
```

Is replaced by 

```swift 
class AClassWithAVeryLongName{
    typealias ACWAVLN =  Self
}
```

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint
```

### Environment

* SwiftLint version 0.52.2 
* Installation method used Homebrew

```yml
opt_in_rules:
 - prefer_self_in_static_references
```

* Which Xcode version are you using 14.3

```swift
class AClassWithAVeryLongName{
    typealias ACWAVLN =  AClassWithAVeryLongName
}
```
